src/components/ChannelList/__tests__/ChannelList.test.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable sonarjs/no-unused-collection */
+import React from 'react';
+import { cleanup, render, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import { Chat } from '../../Chat';
+import ChannelList from '../ChannelList';
+import {
+  getTestClient,
+  useMockedApis,
+  queryChannelsApi,
+  generateMessage,
+  generateChannel,
+  generateUser,
+  getOrCreateChannelApi,
+  dispatchMessageNewEvent,
+  dispatchNotificationAddedToChannelEvent,
+} from '../../../__mocks__';
+import axios from 'axios';
+
+// eslint-disable-next-line no-undef
+afterEach(cleanup);
+
+/**
+ * We are gonna use following custom UI components for preview and list.
+ * If we use ChannelPreviewMessanger or ChannelPreviewLastmessage here, then changes
+ * to those components might endup breaking tests for ChannelList, which will be quite painful
+ * to debug then.
+ */
+const ChannelPreviewComponent = ({ channel, latestMessage }) => (
+  <div role="listitem" data-testid={channel.id}>
+    <div>{channel.data.name}</div>
+    <div>{latestMessage}</div>
+  </div>
+);
+
+const ChannelListComponent = (props) => {
+  const { error, loading, LoadingErrorIndicator, LoadingIndicator } = props;
+  if (error) {
+    return <LoadingErrorIndicator type="Connection Error" />;
+  } else if (loading) {
+    return <LoadingIndicator />;
+  } else {
+    return <div role="list">{props.children}</div>;
+  }
+};
+
+jest.mock('axios');
+
+describe('ChannelList', () => {
+  let chatClientVishal;
+
+  it('should move the channel to top if new message is received', async () => {
+    const channel1 = generateChannel();
+    const channel2 = generateChannel();
+    const channel3 = generateChannel();
+
+    useMockedApis(axios, [queryChannelsApi([channel1, channel2, channel3])]);
+
+    chatClientVishal = await getTestClient({ id: 'vishal' });
```

Comment:
not sure if these tests are running in parallel atm, but I see getTestClient is called twice. does it cause an issue with the mocks if it's moved to `describe` scope?